File: src/main/java/br/com/alura/forum/task/RegisterUnansweredTopicsTask.java
package br.com.alura.forum.task;

import java.util.List;

import br.com.alura.forum.repository.TopicRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

import br.com.alura.forum.model.OpenTopicByCategory;
import br.com.alura.forum.repository.OpenTopicByCategoryRepository;

@Component
public class RegisterUnansweredTopicsTask {
	@Autowired
	private TopicRepository topicRepository;
	@Autowired
	private OpenTopicByCategoryRepository openTopicByCategoryRepository;

	@Scheduled(cron = "0 0 20 * * *")
	public void execute() {
		List<OpenTopicByCategory> topics = this.topicRepository.findOpenTopicsByCategory();
		this.openTopicByCategoryRepository.saveAll(topics);
	}
}

File: src/main/kotlin/br/com/alura/forum/task/RegisterUnansweredTopicsTask.kt
package br.com.alura.forum.task

import br.com.alura.forum.repository.OpenTopicByCategoryRepository
import br.com.alura.forum.repository.TopicRepository
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class RegisterUnansweredTopicsTask(
        private val topicRepository: TopicRepository,
        private val openTopicByCategoryRepository: OpenTopicByCategoryRepository
) {
    @Scheduled(cron = "* * 20 * * *")
    fun execute() {
        val openTopicsByCategory = topicRepository.findOpenTopicsByCategory()
        openTopicByCategoryRepository.saveAll(openTopicsByCategory)
    }
}